### PR TITLE
[MoveosStd] Refactor ObjectRef, allow ObjectRef as transaction argument

### PR DESCRIPTION
**Pull request by jolestar**
## Summary

1. [MoveosStd] Refactor ObjectRef, allow ObjectRef as transaction argument
2. [CLI] verify modules when publishing on the client side.
3. [Example] Refactor simple_blog example via new ObjectRef API.

Part of #1001 

1. Anyone can obtain `&ObjectRef<T>` by ObjectID, and then get `&T`. This can be achieved through the `entry` function arguments or `context::borrow_object(ctx, ObjectID)`.
2. Only the owner can obtain `&mut ObjectRef<T>` by `ObjectID`, and then get `&mut T`. This can be achieved through the `entry` function arguments or `context::borrow_mut_object(ctx,signer,ObjectID)`.
3. If `T` has both `key` and `store` ability, anyone can transfer ownership by  `object_ref::transfer(&mut Object<T>, address)`.
4. The module where `T` is defined can use `context::borrow_mut_object_extend` to obtain any `&mut ObjectRef<T>`, and can transfer ownership by using `object_ref::transfer_extend(&mut Object<T>, address)`.
5. Only the module where `T` is defined can call the `object_ref::remove(ObjectRef<T>)` method to remove the Object.
6. The `object_ref::to_permanent` method is provided to drop `ObjectRef<T>`, but once `ObjectRef<T>` is dropped, the Object cannot be removed from object storage.

### TODO in this PR
1. Fix examples

### TODO in next PR
1. Rename `Object` to `ObjectEntity` and `object.move` to `object_entity.move`.
2. Rename `ObjectRef` to `Object` and `object_ref.move` to `object.move`

### CODE REVIEW

File: moveos/moveos-stdlib/moveos-stdlib/sources/context.move [hunk 1]
```diff
@@ -92,61 +95,58 @@ module moveos_std::context {
...
+        object_ref::as_mut_ref(object_entity)
     }
 
     #[private_generics(T)]
-    /// Remove object from object store, and unpack the Object
-    public fun remove_object<T: key>(self: &mut Context, object_id: ObjectID): (ObjectID, address, T) {
-        let obj = storage_context::remove<T>(&mut self.storage_context, object_id);
-        object::unpack_internal(obj)
+    /// The module of T can borrow mut Object from object store with any object_id
+    public fun borrow_object_mut_extend<T: key>(self: &mut Context, object_id: ObjectID) : &mut ObjectRef<T> {
```
Comment (baichuan3) [inline moveos/moveos-stdlib/moveos-stdlib/sources/context.move:118]: If an object is transferred to another address, the module where the object<T> is defined can still get the write reference of the object through `borrow_object_mut_extend`?

Comment (jolestar) [reply moveos/moveos-stdlib/moveos-stdlib/sources/context.move:118]: yes

File: moveos/moveos-stdlib/moveos-stdlib/sources/object_ref.move [hunk 2]
```diff
@@ -8,48 +8,74 @@ module moveos_std::object_ref {
...
         let ObjectRef{id} = self;
         let object = raw_table::remove_from_global(&id);
-        let (_id, _owner, value) = object::unpack_internal(object);
+        let (_id, _owner, value) = object::unpack(object);
         value
     }
 
+    /// Directly drop the ObjectRef, the object will can not be removed from the object storage.
+    /// If you want to remove the object, please use `remove` function.
+    public fun drop<T: key>(self: ObjectRef<T>) {
```
Comment (jolestar) [inline moveos/moveos-stdlib/moveos-stdlib/sources/object_ref.move:60]: to_unremovable? to_permanent?

File: moveos/moveos-stdlib/moveos-stdlib/sources/object_ref.move [hunk 3]
```diff
@@ -3,53 +3,100 @@
...
+    public fun to_permanent<T: key>(self: ObjectRef<T>) {
+        let ObjectRef{id:_} = self;
+    }
+
+    /// Make the Object shared, Any one can get the &mut ObjectRef<T> from shared object
+    /// The shared object also can be removed from the object storage.
+    public fun to_shared<T: key>(self: ObjectRef<T>) {
+        let obj = raw_table::borrow_mut_from_global<T>(&self.id);
+        object::to_shared(obj);
+        object::transfer_to_system(obj);
```
[the remark this answers is not in the document]
Comment (jolestar) [reply moveos/moveos-stdlib/moveos-stdlib/sources/object_ref.move]: done

**Review by baichuan3 — APPROVED** [review]

**Pull request merged**